### PR TITLE
Updates to test on Python 3.11

### PR DESCRIPTION
**Pull request by pscheidler**

### CODE REVIEW

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [windows-latest, ubuntu-latest, macos-latest]
-        python-version: [3.7, 3.8, 3.9]
+        python-version: [3.7, 3.8, 3.9, 3.10, 3.11]
```
Comment (StokesMIDE) [inline .github/workflows/unit-tests.yml:15]: At least 3.10 needs to be in quotes, so it isn't 3.1

**Review by StokesMIDE — CHANGES_REQUESTED** [review]

**Comment by pscheidler** [comment]
Need to merge in MideTechnology/idelib#129 before the 3.11 tests will pass

File: .github/workflows/unit-tests.yml (hunk 1 above)
Comment (pscheidler) [reply .github/workflows/unit-tests.yml]: Done!

**Review by StokesMIDE — APPROVED** [review]
Tests now passing. There's a deprecation warning regarding a command in the GHA that we should remember to redo later. It's probably in most of our repos as well.

**Comment by pscheidler** [comment]
> Tests now passing. There's a deprecation warning regarding a command in the GHA that we should remember to redo later. It's probably in most of our repos as well.

https://midetech.atlassian.net/browse/ES-496 Josh has fixed these warnings in the firmware test repos, so it should be straightforward

**Pull request merged**